src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -444,11 +396,15 @@ public bool SetValue(TKey key, TValue value, DateTime expirationTime)
             if (_removeExpiredValues && expirationTime < DateTime.UtcNow)
                 return false;
 
+            if (Interlocked.CompareExchange(ref _compactValuesState, ActionQueuedOrRunning, ActionQueuedOrRunning) == ActionQueuedOrRunning)
```

Comment:
Can you help me understand this?

Why do we use ActionQueuedOrRunning for both the value and the comparand?